### PR TITLE
sizes, 64bit

### PR DESCRIPTION
**Pull request by burrbull**
r? @therealprof 

Right use of cluster.size + better support of 64bit registers.

See #299 , partially closed in #325 .

Not tested yet.

### CODE REVIEW

**Comment by burrbull** [comment]
изображение

File: src/util.rs [hunk 1]
```diff
@@ -216,10 +216,16 @@ pub fn access_of(register: &Register) -> Access {
...
 
 /// Turns `n` into an unsuffixed separated hex token
-pub fn hex(n: u32) -> Tokens {
+pub fn hex(n: u64) -> Tokens {
     let mut t = Tokens::new();
-    let (h2, h1) = ((n >> 16) & 0xffff, n & 0xffff);
-    t.append(if h2 != 0 {
+    let (h4, h3, h2, h1) = ((n >> 48) & 0xffff, (n >> 32) & 0xffff, (n >> 16) & 0xffff, n & 0xffff);
+    t.append(if n == 1 {
+        Ident::from("1")
```
Comment (therealprof) [inline src/util.rs]: This special case is weird...

**Comment by therealprof** [comment]
I was kind of hoping that at some point we could get rid of the custom formatting hacks but at the moment there still seems to be no way to generate lint-less long literals with the built-in formatting mechanics...

**Comment by burrbull** [comment]
Don't merge. Something wrong.

File: src/util.rs (hunk 1 above)
Comment (burrbull) [reply src/util.rs]: What are you offering?

Comment (therealprof) [reply src/util.rs]: Getting rid of it? I'd fully expect this function to always return a real hex literal, not a hex literal or "1".

Comment (burrbull) [reply src/util.rs]: Or you about "1"? I can delete this.

Comment (therealprof) [reply src/util.rs]: Yes, I don't like a function called `hex` to treat the number 1 special and not turn it into hex. ;)

**Comment by burrbull** [comment]
Fixed.

**Comment by therealprof** [comment]
Looks great. Can you pleasse add a CHANGELOG entry, too? That way we don't have to hunt down all changes after the fact. ;)

**Comment by burrbull** [comment]
Done.

**Pull request merged**